README.md

```diff
@@ -24,76 +24,276 @@ undirected graphs, and electrical circuits.
+* NFA
+* Directed graph
+* Graph
+* Binary tree
 
-This is how you would write your DFA:
+## This is how you would write your DFA:
 
 Starting State:
 
 	let startState = X_i
 
-States (integers):
+States:
 
 	let states = [X_1,X_2,X_3,...]
 
-Labels (characters at the moment):
+Labels:
 
-	let alfabeth = [Symbol_1, Symbol_2,..]
+	let alfabeth = [L_1, L_2,..]
 
 Transitions:
 
-	let transitions = [(X_i,X_j,Symbol_h),...]
+	let transitions = [(X_i,X_j,L_h),...]
 
 Accepted States:
 
 	let acceptStates = [X_i,X_j,...]
+	
+To visualize the DFA in action, please write your input in the form of an array of labels as follows:
+
+	let input = [L_1,L_1,L_2,L_1,...]
+	
+There are no data type requirements, thus you would need to write equality functions for the states (eqv) and labels (eql) (X and L in the above examples). The equality functions get 2 inputs and returns either **true** or **false** (true if the two states/labels are equal and false otherwise). 
+
+* For example, if your states were integers and your labels were chars, you could do:
+
+		let eqv = lam s1. lam s2.
+			eqi s1 s2
+		let eql = lam s1. lam s2.
+			eqchar s1 s2
+
+
 
 To construct a DFA use this function:
 
-	let your_dfa = dfaConstr states transitions alfabeth startState acceptStates
+	let your_dfa = dfaConstr states transitions
+    alfabeth startState acceptStates eqv eql
 
-To create the visualizer, make sure to use either of this two:
 
-	let visual = dfaVisualNoInput your_dfa in
-	print visual
+## This is how you would write your NFA:
+A NFA works the same as a DFA, just replace "dfa" with "nfa". The transitions labels do not need to be unique for a state.
 
-or if you want to see the input:
+## This is how you would write your directed graph:
 
-	let visual = dfaVisual your_dfa input in
-	print visual
+To start, create an empty digraph. This can be done with:
+	
+	digraphEmpty eqv eql
 
-You can start the server for watching your file using this command and sourcing your **.mc** file (this would be if your file is in the root directory of the project):
+There are no data type requirements, thus you would need to write equality functions for the vertices (eqv) and labels (eql). The equality functions get 2 inputs and returns either **true** or **false** (true if the two states/labels are equal and false otherwise). 
+
+For example, if your vertices were integers and your labels were charecters, you could do:
+
+	let empty = digraphEmpty eqi eqchar
+
+To add vertexes and edges to the graph use these commands:
+
+	digraphAddVertex v g
+	digraphAddEdge v1 v2 l g
+
+Where v, v1 and v2 are vertices, l is a label for an edge and g is the previous digraph. 
+
+## This is how you would write your graph:
+A graph works the same was as the digraph, just replace `digraph` with `graph` in the functions.
+
+## This is how you would write your binary tree:
+To create a binary tree the constructor BTree can be used.
+
+	BTree (Node(value, Nil()/Node()/Leaf(),Nil()/Node()/Leaf())) 
```

Comment:
This example has strange syntax. What is `/`? This is not how it looks in the example code.
Again, it would be good if you just include a small example (complete) and explain it.

Reply:
This has been updated in the README